spring-cloud-task-core/src/test/java/org/springframework/cloud/task/configuration/RepositoryTransactionManagerConfigurationTests.java

```diff
@@ -0,0 +1,237 @@
+import org.springframework.boot.jdbc.DataSourceBuilder;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.cloud.task.listener.TaskLifecycleListener;
+import org.springframework.cloud.task.repository.TaskExecution;
+import org.springframework.cloud.task.repository.TaskRepository;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.datasource.DataSourceTransactionManager;
+import org.springframework.test.jdbc.JdbcTestUtils;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.support.DefaultTransactionStatus;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * @author Michael Minella
+ */
+public class RepositoryTransactionManagerConfigurationTests {
+
+	@Test
+	public void testZeroCustomTransactionManagerConfiguration() {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+			.withConfiguration(
+				AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+					SimpleTaskAutoConfiguration.class,
+					ZeroTransactionManagerConfiguration.class))
+			.withPropertyValues("application.name=transactionManagerTask");
+
+		applicationContextRunner.run((context) -> {
+			DataSource dataSource = context.getBean("dataSource", DataSource.class);
+
+			int taskExecutionCount = JdbcTestUtils
+				.countRowsInTable(new JdbcTemplate(dataSource), "TASK_EXECUTION");
+
+			assertThat(taskExecutionCount).isEqualTo(1);
+		});
+	}
+
+	@Test
+	public void testSingleCustomTransactionManagerConfiguration() {
+		testConfiguration(SingleTransactionManagerConfiguration.class);
+	}
+
+	@Test
+	public void testMultipleCustomTransactionManagerConfiguration() {
+		testConfiguration(MultipleTransactionManagerConfiguration.class);
+	}
+
+	private void testConfiguration(Class configurationClass) {
+		ApplicationContextRunner applicationContextRunner = new ApplicationContextRunner()
+			.withConfiguration(
+				AutoConfigurations.of(PropertyPlaceholderAutoConfiguration.class,
+					SimpleTaskAutoConfiguration.class,
+					configurationClass))
+			.withPropertyValues("application.name=transactionManagerTask");
+
+		applicationContextRunner.run((context) -> {
+			DataSource dataSource = context.getBean("dataSource", DataSource.class);
+
+			int taskExecutionCount = JdbcTestUtils
+				.countRowsInTable(new JdbcTemplate(dataSource), "TASK_EXECUTION");
+
+			// Verify that the create call was rolled back
+			assertThat(taskExecutionCount).isEqualTo(0);
+
+			// Execute a new create call so that things close cleanly
+			TaskRepository taskRepository = context.getBean("taskRepository",
+				TaskRepository.class);
+
+			TaskExecution taskExecution = taskRepository
```

Comment:
We can move the taskExecution declaration and initialization to the line below because we overwrite it on line 101.

Reply:
Actually, we should use the `taskExecution` to get the executionId on line 101 instead of hard coding the id to 2.  I changed the test and forgot to clean that up.